src/libpriv/rpmostree-passwd-util.c

```diff
@@ -1114,9 +1115,15 @@ rpmostree_passwd_compose_prep (int              rootfs_dfd,
   if (!found_passwd_data &&  found_groups_data)
     return glnx_throw (error, "Configured to migrate passwd data from commit, and group data from JSON");
 
-  if (found_passwd_data || !previous_root)
+  if (found_passwd_data || !previous_checksum)
     return TRUE; /* Nothing to do */
 
+  g_assert (repo);
```

Comment:
This should be guarded with `if (previous_checksum)` right?

Reply:
We do `!previous_checksum` just above. :)